stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/catalog/uicomponents/customattachments/composer/ChannelsActivity.kt

```diff
@@ -0,0 +1,91 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.guides.catalog.uicomponents.customattachments.composer
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.guides.catalog.uicomponents.customattachments.composer.factory.DateAttachmentFactory
+import io.getstream.chat.android.guides.catalog.uicomponents.customattachments.composer.factory.DateAttachmentPreviewFactory
+import io.getstream.chat.android.guides.catalog.uicomponents.customattachments.composer.factory.QuotedDateAttachmentFactory
+import io.getstream.chat.android.ui.ChatUI
+import io.getstream.chat.android.ui.channel.ChannelListActivity
+import io.getstream.chat.android.ui.channel.ChannelListFragment
+import io.getstream.chat.android.ui.message.composer.attachment.AttachmentPreviewFactoryManager
+import io.getstream.chat.android.ui.message.composer.attachment.factory.FileAttachmentPreviewFactory
+import io.getstream.chat.android.ui.message.composer.attachment.factory.ImageAttachmentPreviewFactory
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.AttachmentFactoryManager
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.DefaultQuotedAttachmentMessageFactory
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.QuotedAttachmentFactoryManager
+
+/**
+ * An Activity representing a self-contained channel list screen with custom attachment factories.
+ */
+class ChannelsActivity : ChannelListActivity(), ChannelListFragment.ChannelListItemClickListener {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        ChatUI.attachmentFactoryManager = AttachmentFactoryManager(
+            attachmentFactories = listOf(
+                DateAttachmentFactory()
+            )
+        )
+        ChatUI.attachmentPreviewFactoryManager = AttachmentPreviewFactoryManager(
+            attachmentPreviewFactories = listOf(
+                DateAttachmentPreviewFactory(),
+                ImageAttachmentPreviewFactory(),
+                FileAttachmentPreviewFactory(),
+            )
+        )
+        ChatUI.quotedAttachmentFactoryManager = QuotedAttachmentFactoryManager(
+            quotedAttachmentFactories = listOf(
+                QuotedDateAttachmentFactory(),
+                DefaultQuotedAttachmentMessageFactory()
+            )
+        )
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        // Reset to the default values not to affect other samples
```

Comment:
Hope this cleanup won't distract the users.

Reply:
Maybe we can create a helper object, do the cleanup there and just call the function? We are doing those cleanups in multiple places